server/src/main/java/org/elasticsearch/index/mapper/ObjectMapper.java

```diff
@@ -820,6 +820,9 @@ public void write(XContentBuilder b) throws IOException {
 
         @Override
         public boolean setIgnoredValues(Map<String, List<IgnoredSourceFieldMapper.NameValue>> objectsWithIgnoredFields) {
+            if (objectsWithIgnoredFields == null || objectsWithIgnoredFields.isEmpty()) {
+                return false;
+            }
             ignoredValues = objectsWithIgnoredFields.get(name());
```

Comment:
maybe this can be `objectsWithIgnoredFields.remove(name())`? Then when the map is empty for deeply nested structures we stop the looping of empty maps? This would help is there is one or a few ignored sources?

Reply:
Good call, done.